src/index.js

```diff
@@ -221,33 +195,42 @@ class IpfsRepo {
   exists (callback) {
     this.version.exists(callback)
   }
+}
 
-  /**
-   * Set the api address, by writing it to the `/api` file.
-   *
-   * @param {Multiaddr} addr
-   * @param {function(Error)} callback
-   * @returns {void}
-   */
-  setApiAddress (addr, callback) {
-    this._fsStore.put(apiFile, Buffer.from(addr.toString()), callback)
-  }
+module.exports = IpfsRepo
 
-  /**
-   * Returns the registered API address, according to the `/api` file in this respo.
-   *
-   * @param {function(Error, Mulitaddr)} callback
-   * @returns {void}
-   */
-  apiAddress (callback) {
-    this._fsStore.get(apiFile, (err, rawAddr) => {
-      if (err) {
-        return callback(err)
-      }
+function ignoringAlreadyOpened (cb) {
+  return (_err) => {
+    let err = _err
```

Comment:
We currently don't have unified error handling on the datastore interfaces. These errors come directly from them which is why have to make string matching right now. I agree it would be better to have subclasses of `Error` + error codes for checking these but these need to be implemented first in all the datastores.